src/App/Document.cpp

```diff
@@ -4189,13 +4189,20 @@ const std::vector<DocumentObject*>& Document::getObjects() const
     return d->objectArray;
 }
 
-
 std::vector<DocumentObject*> Document::getObjectsOfType(const Base::Type& typeId) const
+{
+    return getObjectsOfType({typeId});
+}
+
+std::vector<DocumentObject*> Document::getObjectsOfType(const std::vector<const Base::Type>& types) const

```

Comment:
```suggestion
std::vector<DocumentObject*> Document::getObjectsOfType(const std::initializer_list<const Base::Type>& types) const
```

That should work too? I don't think that vector is required as input struct, `initializer_list` seems to be better suited for that kind of job.

Reply:
`std::vector` does allocation on stack and is used for storing arrays of changing sizes, i.e. it can grow and shrink - reallocating memory in the process.
 
`std::initializer_list` on the other hand is lightweight proxy for constant array, basically the `{ x, y, z }` syntax is `std::initializer_list` and so you construct first initializer_list and then implicitly cast it into vector in your code.

Downside here is that `std::initializer_list` is constant, so you cannot construct it via code.

Maybe @FreeCAD/code-quality-working-group will have some opinion here? I don't have strong opinion on that one, but in general I prefer to use simpler structs as inputs and vector is quite advanced one.

Reply:
So I leave std::vector then right?

Reply:
I'm fine with that - we don't have guidelines on how to approach situations like this and this particular function should be implemented using variadic templates anyway. I (or someone else) will later create PR with the templated version and then revisit it if needed.